chap1-introduction.tex

```diff
@@ -23,84 +23,69 @@ \chapter{Introduction}\label{ch:intro}
-%while there is a great deal of research on the topic of efficiently producing proofs in a fully automated way~ these tools, at least at present, cannot hope to solve the challenges
-%\todo{integrate stuff from \textcite{Efficiency1994Boulton} such as ``For example, much of the research in the field of automated reasoning is concerned with producing proofs more efficiently''}
-%\todo{Related to this is work on sound interfaces to external tools, also an enduring topic: \textcite{TrustedSlind}}
 
-\minortodo{work on this paragraph}
-Why is this problem worthy of study?
-This thesis argues that the problem of \emph{compile-time performance} or \emph{proof-checking performance} is both significantly different from the problem of performance of typical programs and that it is nontrivial.
-While many papers mention performance, obliquely or otherwise, and some are even driven by performance concerns of a particular algorithm or part of the system,~\cites[p.~1382]{gonthier2008formal}{Efficiency1994Boulton}{Proving2005Benjamin}{Idris2Faster2020Brady}{Recognizing1989Benanav}{mechanical1990Pierce}{CelikETAL17iCoq}{PalmskogETAL18piCoq}{vmcompute}{thesis-nogin}{Idris2Faster2020Brady}, we have not found any that investigate the performance problems that arise \emph{asymptotically}, when proof assistants are used to verify programs at large scale.
-% ``While we tackled this project mainly to explore the capabilities of a modern formal proof system—at first, to benchmark speed—'' - gonthier2008formal
-% ``We present a new implementation of a reflexive tactic whichsolves equalities in a ring structure inside the Coq system.The efficiencyis improved to a point that we can now prove equalities that were previ-ously beyond reach.'' - Proving2005Benjamin
+However, in all of these tools, there is enormous proof overhead over the lines of code being verified. (Insert graph) 
 
-\todonz{Seems fine ... but is it really unique to proofs?  C++ compilers in the 90s had the same problems}
+To decrease the marginal overhead of proof generation, we may automate the proof generation itself. 
+Drawing experience from case studies in category theory~\cite{category-coq-experience}, parsers~\cite{jgross-masters-thesis}, and generation of low-level cryptographic code~\cite{FiatCryptoSP19}, we present the difficulties in automation of proof generation. 
+In the quest to build automated proof generation tools, we encountered several performance bottlenecks in proof-checking. In this thesis, we present design principles in \autoref{part:design} to avoid these performance bottlenecks, and insights about the proof-checking system that the performance bottlenecks reveal. We present a research prototype of a tool and methodology for achieving acceptable performance at scale in \autoref{part:rewriting}. 
```

Comment:
```suggestion
In the quest to build automated proof generation tools, we encountered several performance bottlenecks in proof-checking.
In this thesis, we present design principles in \autoref{part:design} to avoid these performance bottlenecks, and insights about the proof-checking system that the performance bottlenecks reveal.
We present a research prototype of a tool and methodology for achieving acceptable performance at scale in \autoref{part:rewriting}. 
```